### DISCUSSION TITLE
Fallback for missing resolvers

### DISCUSSION THREAD

**Discussion by eduludi** [prevwong/craft.js#276] [Ideas]
Is there a way of providing a fallback for nodes that are not listed in the resolver?

If a node's name doesn't match one of the resolvers,  an error is thrown:

```
Uncaught TypeError: Cannot read property 'name' of undefined
```

I'm thinking in the case when a change or rename in the resolver for a node, and this will cause all pages/documents that still reference to the old resolver will crash the editor. 

In these cases a graceful fallback may help, at least for the developers to know what's going on (where the missing node's name may be displayed).

**Comment by ankri** [comment]
With a coming version (`next` branch) we will list all the Resolvers that could not be found.

But there is no graceful fallback. 

But you could roll your own:

```ts
// this is from our codebase. ComponentLibrary holds all our components and exposes a method to findComponentsById
// if the component cannot be found it throws an error (thus the try catch block)
// you can alter it to your needs an way you store your resolvers
export const findComponentsWithoutResolvers = (
  nodes: SerializedNodes
): SerializedNode[] => {
  return Object.entries(nodes)
    .map(([key, node]) => {
      try {
        if (typeof node.type === 'string') {
          ComponentLibrary.getInstance().findComponentById(node.type);
        } else {
          ComponentLibrary.getInstance().findComponentById(
            node.type.resolvedName
          );
        }
        return null;
      } catch (e) {
        return node;
      }
    })
    .filter((node) => node !== null);
};
```

you can call it with `findComponentsWithoutResolvers(JSON.parse(data) as SerializedNodes)` to get a list of all missing resolvers and then add a Fallback renderer for those.

**Reply by pcjaarvis** [reply]
any updates on this?

**Reply by hugominas** [reply]
This is an issue when developing components, it would be great to have this

**Comment by addlistener** [comment]
maybe a custom property getter for the resolver object?
